io.asgardio.java.oidc.sdk/src/main/java/io/asgardio/java/oidc/sdk/OIDCManagerImpl.java

```diff
@@ -0,0 +1,307 @@
+
+            if (!authorizationResponse.indicatesSuccess()) {
+                handleErrorAuthorizationResponse(authorizationResponse);
+                return false;
+            } else {
+                successResponse = authorizationResponse.toSuccessResponse();
+                authorizationCode = successResponse.getAuthorizationCode();
+            }
+            tokenRequest = getTokenRequest(authorizationCode);
+            tokenResponse = getTokenResponse(tokenRequest);
+
+            if (!tokenResponse.indicatesSuccess()) {
+                handleErrorTokenResponse(tokenRequest, tokenResponse);
+                return false;
+            } else {
+                handleSuccessTokenResponse(tokenResponse, authenticationInfo);
+                return true;
+            }
+        } catch (com.nimbusds.oauth2.sdk.ParseException | SSOAgentServerException | IOException e) {
+            logger.error(e.getMessage(), e);
+            return false;
+        }
+    }
+
+    private void handleSuccessTokenResponse(TokenResponse tokenResponse, AuthenticationInfo authenticationInfo)
+            throws SSOAgentServerException {
+
+        AccessTokenResponse successResponse = tokenResponse.toSuccessResponse();
+        AccessToken accessToken = successResponse.getTokens().getAccessToken();
+        RefreshToken refreshToken = successResponse.getTokens().getRefreshToken();
+        String idToken;
+        try {
+            idToken = successResponse.getCustomParameters().get(SSOAgentConstants.ID_TOKEN).toString();
+        } catch (NullPointerException e) {
+            logger.log(Level.ERROR, "id_token is null.");
+            throw new SSOAgentServerException(SSOAgentConstants.ErrorMessages.ID_TOKEN_NULL.getMessage(),
+                    SSOAgentConstants.ErrorMessages.ID_TOKEN_NULL.getCode(), e);
+        }
+        try {
+            JWTClaimsSet claimsSet = SignedJWT.parse(idToken).getJWTClaimsSet();
+            User user = new User(claimsSet.getSubject(), getUserAttributes(idToken));
+
+            authenticationInfo.setIdToken(JWTParser.parse(idToken));
+            authenticationInfo.setUser(user);
+            authenticationInfo.setAccessToken(accessToken);
+            authenticationInfo.setRefreshToken(refreshToken);
+        } catch (ParseException e) {
+            throw new SSOAgentServerException(SSOAgentConstants.ErrorMessages.ID_TOKEN_PARSE.getMessage(),
+                    SSOAgentConstants.ErrorMessages.ID_TOKEN_PARSE.getCode(), e);
+        }
+    }
+
+    private void handleErrorTokenResponse(TokenRequest tokenRequest, TokenResponse tokenResponse) {
+
+        TokenErrorResponse errorResponse = tokenResponse.toErrorResponse();
+        JSONObject requestObject = requestToJson(tokenRequest);
+        JSONObject responseObject = errorResponse.toJSONObject();
+        logger.log(Level.INFO, "Request object for the error response: ", requestObject);
+        logger.log(Level.INFO, "Error response object: ", responseObject);
```

Comment:
Is the log level correct?

Reply:
Resolved with https://github.com/asgardio/asgardio-java-oidc-sdk/pull/1/commits/87be1ad8b44e103c81d6ce8a6ed22d4f6fc4f854.